### PR TITLE
feat: add sorted set commands

### PR DESCRIPTION
**Pull request by brayniac**
Adds the sorted set commands for the Momento protocol to the SDK.

### CODE REVIEW

**Comment by kvcache** [comment]
separately from the auth token thing there appear to be some other issues:
```
---- src/simple_cache_client.rs - simple_cache_client::SimpleCacheClient::sorted_set_get_rank (line 1954) stdout ----
error[E0609]: no field `value` on type `Option<u64>`
  --> src/simple_cache_client.rs:1963:87
   |
11 | match momento.sorted_set_get_rank(&cache_name, "test sorted set", "element a").await?.value {
   |                                                                                       ^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0609`.
Couldn't compile the test.
---- src/simple_cache_client.rs - simple_cache_client::SimpleCacheClient::sorted_set_get_score (line 2028) stdout ----
error[E0599]: no method named `as_string` found for reference `&'static str` in the current scope
  --> src/simple_cache_client.rs:2043:64
   |
17 |     let score = score.map(|s| format!("{s}")).unwrap_or("None".as_string());
   |                                                                ^^^^^^^^^ help: there is a method with a similar name: `as_ptr`

```

File: src/simple_cache_client.rs [hunk 1]
```diff
@@ -1859,6 +1862,397 @@ impl SimpleCacheClient {
...
+    /// * `set_name` - name of the set.
+    /// * `order` - specify ascending or descending order
+    /// * `limit` - optionally limit the number of results returned
+    ///
+    /// # Example
+    /// ```
+    /// # fn main() -> momento_test_util::DoctestResult {
+    /// # momento_test_util::doctest(|cache_name, auth_token| async move {
+    /// use std::time::Duration;
+    /// use momento::{Order, SimpleCacheClientBuilder};
```
Comment (kvcache) [inline src/simple_cache_client.rs:1882]: `momento::Order` is a pretty aggressive namespace. I think you want to scope it under something sorted set related.

**Review by kvcache — COMMENTED** [review]
Looks fine overall. Got a build break and 1 published namespace concern

**Comment by brayniac** [comment]
@kvcache - I think this is now ready for review after merging main into the branch.

**Review by kvcache — APPROVED** [review]
looks good, thanks!

**Pull request merged**